src/Simplify.cpp

```diff
@@ -3205,6 +3233,18 @@ void simplify_test() {
     check((x - cast(Float(64), 0.5f)) * (x - cast(Float(64), 0.5f)),
           (x + Expr(-0.5)) * (x + Expr(-0.5)));
 
+    check(cast(Int(64).vector_of(3), ramp(x, 2, 3)),
+          ramp(cast(Int(64), x), cast(Int(64), 2), 3));
+
+    check(cast(Int(64), x + 1) - cast(Int(64), x), cast(Int(64), 1));
+    // It is not safe to perform the same simplification on floats, as
+    // it may enable associativity to change:
```

Comment:
We mess with floating point associativity all the time. The reason not to apply these rules to floats is that it might not be an optimization. E.g. cast<float>(some_double) - cast<float>(some_double) might be cheaper than cast<float>(some_double - some_double).